### PR TITLE
Waitp 1288 mobile overlay selector

### PR DESCRIPTION
**Pull request by acdunham**
### Changes:
- Added overlay library list, title, and datepicker to mobile overlay selector
- Styled mobile overlay selector
- Tweaked styling around map so that attribution is visible on mobile
---

### Screenshots:
Screenshot 2023-07-21 at 10 58 40 am

Screenshot 2023-07-21 at 10 58 50 am

Screenshot 2023-07-21 at 10 59 29 am
Screenshot 2023-07-21 at 11 00 18 am

### CODE REVIEW

File: packages/tupaia-web/src/features/Map/MapOverlaySelector/DesktopMapOverlaySelector.tsx [hunk 1]
```diff
@@ -53,6 +54,15 @@ const Container = styled(MaxHeightContainer)`
   pointer-events: auto;
 `;
 
+const TitleWrapper = styled.div`
+  padding: 1rem;
+  background-color: ${({ theme }) => theme.overlaySelector.overlayNameBackground};
+  // Add padding between the title and the date picker when both are present
```
Comment (tcaiger) [inline packages/tupaia-web/src/features/Map/MapOverlaySelector/DesktopMapOverlaySelector.tsx:60]: thanks for the comment

File: packages/tupaia-web/src/features/Map/MapOverlaySelector/MobileMapOverlaySelector.tsx [hunk 2]
```diff
@@ -44,40 +55,77 @@ const OverlayLibraryHeader = styled.span`
...
 
+const OverlayListWrapper = styled.div`
+  padding: 1rem;
+`;
+
 const OverlayMenu = styled.div<{
   $expanded: boolean;
 }>`
-  height: ${({ $expanded }) => ($expanded ? '100%' : '0')};
+  height: ${({ $expanded }) => ($expanded ? '95vh' : '0')};
```
Comment (tcaiger) [inline packages/tupaia-web/src/features/Map/MapOverlaySelector/MobileMapOverlaySelector.tsx:65]:
```suggestion
height: ${({ $expanded }) => ($expanded ? `calc(100vh - ${TOP_BAR_HEIGHT_MOBILE}px)` : '0')};
```
I think the 95vh is a magic number and not always going to fit the space. We could move the `TOP_BAR_HEIGHT_MOBILE` to the constants file and use it to calculate the height.

**Review by tcaiger — CHANGES_REQUESTED** [review]
Looks good - nice pure css changes :) I just have one requested change re the expanded height. thanks

File: packages/tupaia-web/src/features/Map/MapLegend/MapLegend.tsx [hunk 3]
```diff
@@ -37,7 +37,7 @@ export const MapLegend = ({ hiddenValues, setValueHidden }: LegendProps) => {
   const selectedOverlay = urlSearchParams.get(URL_SEARCH_PARAMS.MAP_OVERLAY);
   const { data: overlayReportData } = useMapOverlayReport();
 
-  if (!selectedOverlay) {
+  if (!selectedOverlay || !overlayReportData) {
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Map/MapLegend/MapLegend.tsx:40]: This is to stop the legend on mobile showing when there is no legend to display

**Review by tcaiger — APPROVED** [review]
LGTM :)

**Pull request merged**